src/wp-admin/includes/misc.php

```diff
@@ -1641,3 +1641,133 @@ function wp_check_php_version() {
 
 	return $response;
 }
+
+/**
+ * Creates and returns the markup for an admin notice.
+ *
+ * @since 6.3.0
+ *
+ * @param string $message The message.
+ * @param array  $args {
+ *     Optional. An array of arguments for the admin notice. Default empty array.
+ *
+ *     @type string   $type               Optional. The type of admin notice.
+ *                                        For example, 'error', 'success', 'warning', 'info'.
+ *                                        Default empty string.
+ *     @type bool     $dismissible        Optional. Whether the admin notice is dismissible. Default false.
+ *     @type string   $id                 Optional. The value of the admin notice's ID attribute. Default empty string.
+ *     @type string[] $additional_classes Optional. A string array of class names. Default empty array.
+ *     @type bool     $paragraph_wrap     Optional. Whether to wrap the message in paragraph tags. Default true.
+ * }
+ */
+function wp_get_admin_notice( $message, $args ) {
+	$defaults = array(
+		'type'               => '',
+		'dismissible'        => false,
+		'id'                 => '',
+		'additional_classes' => array(),
+		'paragraph_wrap'     => true,
+	);
+
+	$args = wp_parse_args( $args, $defaults );
+
+	/**
+	 * Filters the arguments for an admin notice.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param array  $args    The arguments for the admin notice.
+	 * @param string $message The message for the admin notice.
+	 */
+	$args    = apply_filters( 'wp_admin_notice_args', $args, $message );
+	$id      = '';
+	$classes = 'notice';
+
+	if ( is_string( $args['id'] ) ) {
+		$trimmed_id = trim( $args['id'] );
+
+		if ( '' !== $trimmed_id ) {
+			$id = 'id="' . esc_attr( $trimmed_id ) . '" ';
+		}
+	}
+
+	if ( is_string( $args['type'] ) ) {
+		$type = trim( $args['type'] );
+
+		if ( '' !== $type ) {
+			if ( 'updated' === $type ) {
+				$type = 'success';
+			}
```

Comment:
Thinking of removing this. It's only based on a conversion in [settings_errors()](https://github.com/wordpress/wordpress-develop/blob/d8731e613a8e7a92678b36dd75a6103b7dc8138b/src/wp-admin/includes/template.php#L1985), which could just be converted when implementing this function in `settings_errors()`. Keeping it here will cause confusion when someone wants a `notice-updated` class name, but get `notice-success` instead.

@joedolson What do you think?

Reply:
I agree. I think it makes more sense to have the customization happen in the place that needs it, rather than a semi-magical alteration of value in the handler function. By putting it in `settings_errors()`, it's clear why it's happening, where it's kind of mystifying as an automatic change. 

(It would benefit from a code comment in `settings_errors()`, perhaps.)

Reply:
Updated in [05def7e](https://github.com/WordPress/wordpress-develop/pull/4119/commits/05def7e6643c1303cb5045d9ff739d29570f1764).